File: assignment_1/app/src/main/java/com/example/assignment_1/activities/MainActivity.java
package com.example.assignment_1.activities;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;

import com.example.assignment_1.R;

public class MainActivity extends AppCompatActivity {

    private Button gradesButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setupUi();
        setupAction();
    }

    private void setupAction() {
        this.gradesButton.setOnClickListener((view)->navigateToGrades());
    }

    private void setupUi() {

        gradesButton = findViewById(R.id.button_grade);
    }

    private void navigateToGrades(){
        Intent myIntent = new Intent(getApplicationContext(), GradeActivity.class);
        startActivity(myIntent);
    }
}

File: Assignment2/app/src/main/java/com/example/assignment2/activities/MainActivity.kt
package com.example.assignment2.activities

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.example.assignment2.R
import com.example.assignment2.helpers.SharedPreferenceHelper

class MainActivity : AppCompatActivity() {

    private lateinit var sharedPreferenceHelper: SharedPreferenceHelper
    private lateinit var profileButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        sharedPreferenceHelper = SharedPreferenceHelper(this@MainActivity)

        profileButton = findViewById(R.id.save_button)

        profileButton.setOnClickListener {goToProfileActivity()}
    }

    override fun onStart() {
        super.onStart()

        val profile = sharedPreferenceHelper.profile

        if (profile == null)
            goToProfileActivity()
        else
            profileButton.text = "${profile.name}\'s profile"
    }

    private fun goToProfileActivity() {
        val intent = Intent(this@MainActivity, ProfileActivity::class.java)
        startActivity(intent)
    }
}
